compiler/rustc_mir/src/transform/validate.rs

```diff
@@ -13,38 +12,32 @@ use rustc_middle::mir::{
 use rustc_middle::ty::relate::{Relate, RelateResult, TypeRelation};
 use rustc_middle::ty::{self, ParamEnv, Ty, TyCtxt};
 
-#[derive(Copy, Clone, Debug)]
-enum EdgeKind {
-    Unwind,
-    Normal,
+pub fn validate_body(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>, when: &str) {
+    validate_body_during_phase(tcx, body, body.phase, when);
 }
 
-pub struct Validator {
-    /// Describes at which point in the pipeline this validation is happening.
-    pub when: String,
-    /// The phase for which we are upholding the dialect. If the given phase forbids a specific
-    /// element, this validator will now emit errors if that specific element is encountered.
-    /// Note that phases that change the dialect cause all *following* phases to check the
-    /// invariants of the new dialect. A phase that changes dialects never checks the new invariants
-    /// itself.
-    pub mir_phase: MirPhase,
-}
+pub fn validate_body_during_phase(
```

Comment:
This function seems to only be used through `validate_body`.

Reply:
I'm aware. It'll be useful later.